trial/src/Trial.hs

```diff
@@ -380,7 +423,10 @@ unTag :: TaggedTrial tag a -> Trial tag a
 unTag (Fiasco e)          = Fiasco e
 unTag (Result e (tag, a)) = Result (DL.snoc e tag) a
 
-{- |
+{- | Convenient instance to convert record fields of type
+'TaggedTrial' to 'Trial' by appending field names to the history. This
+instance automatically combines tags and record field names into human
+readable message, so the resulting history has more context.
```

Comment:
We will need to add a proper usage example later :) Could you please add a TODO?